### PR TITLE
switch to skopt

### PR DESCRIPTION
**Pull request by inventormc**
Change bayesian search to SkOpt backend to allow for searching over discrete integers.

### CODE REVIEW

File: examples/discrete_bayesian.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+digits = datasets.load_digits()
+X = digits.data
+y = digits.target
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2)
+
+space = {'n_estimators': (100,200),
+        'min_samples_split': (2, 10),
+        'min_samples_leaf': (1, 5)}
```
Comment (richardliaw) [inline examples/discrete_bayesian.py:16]: can you demonstrate something using a mixture of discrete and continuous?

File: tune_sklearn/tune_search.py [hunk 2]
```diff
@@ -141,10 +143,10 @@ class TuneSearchCV(TuneBaseSearchCV):
         max_iters (int): Indicates the maximum number of epochs to run for each
             hyperparameter configuration sampled (specified by ``n_iter``).
             This parameter is used for early stopping. Defaults to 10.
-        search_optimization ("random", "bayesian", or :obj:`BayesOptSearch`):
+        search_optimization ("random" or "bayesian"):
             If "random", uses randomized search over the
-            ``param_distributions``. If "bayesian" or ``BayesOptSearch``,
-            uses Bayesian optimization to search for hyperparameters.
+            ``param_distributions``. If "bayesian", uses
+            Bayesian optimization to search for hyperparameters.
```
Comment (richardliaw) [inline tune_sklearn/tune_search.py:149]: uses Bayesian Optimization from scikit-optimize (link to scikit-optimize)

File: tune_sklearn/tune_search.py [hunk 3]
```diff
@@ -50,15 +52,15 @@ class TuneSearchCV(TuneBaseSearchCV):
...
-            Represents search space over parameters of
-            the provided estimator.
+            For Bayesian search: dictionary with parameter names (string)
+            as keys. Values can be
+
+            - a (lower_bound, upper_bound) tuple (for Real
+              or Integer dimensions),
+            - a (lower_bound, upper_bound, "prior") tuple
+              (for Real dimensions),
+            - as a list of categories (for Categorical dimensions), or
```
Comment (richardliaw) [inline tune_sklearn/tune_search.py:62]: you'll want to link to scikit-optimize documentation somehow

Comment (anthonyhsyu) [reply tune_sklearn/tune_search.py:62]: https://scikit-optimize.github.io/stable/modules/generated/skopt.Optimizer.html#skopt.Optimizer

File: README.md [hunk 4]
```diff
@@ -82,7 +82,7 @@ print("Sklearn Accuracy:", accuracy)
 
 #### TuneSearchCV
 
-`TuneSearchCV` uses randomized search over the distribution by default, but can do Bayesian search as well by specifying the `search_optimization` parameter as shown here. You need to run `pip install bayesian-optimization` for this to work. More details in [Tune Documentation](https://docs.ray.io/en/latest/tune-searchalg.html#bayesopt-search).
+`TuneSearchCV` uses randomized search over the distribution by default, but can do Bayesian search as well by specifying the `search_optimization` parameter as shown here. You need to run `pip install scikit-optimize` for this to work. More details in [Tune Documentation](https://docs.ray.io/en/latest/tune-searchalg.html#bayesopt-search).
```
[the remark this answers is not in the document]
Comment (inventormc) [reply README.md:85]: Oh right good catch

File: tune_sklearn/tune_search.py [hunk 5]
```diff
@@ -71,7 +76,6 @@ class TuneSearchCV(TuneBaseSearchCV):
               of schedulers are currently supported. The scheduler will only be
               used if the estimator supports partial fitting
             - If None or False, early stopping will not be used.
-
```
Comment (richardliaw) [inline tune_sklearn/tune_search.py:74]: be careful not to remove these lines

**Pull request merged**